src/decoder/ifd.rs

```diff
@@ -332,171 +332,206 @@ impl Entry {
+                Type::SBYTE => Signed(i32::from(self.offset[0] as i8)),
+                Type::UNDEFINED => Byte(self.offset[0]),
+                Type::SHORT => Unsigned(u32::from(self.r(bo).read_u16()?)),
+                Type::SSHORT => Signed(i32::from(self.r(bo).read_i16()?)),
+                Type::LONG => Unsigned(self.r(bo).read_u32()?),
+                Type::SLONG => Signed(self.r(bo).read_i32()?),
+                Type::FLOAT => Float(self.r(bo).read_f32()?),
+                Type::ASCII => {
+                    if self.offset[0] == 0 {
+                        Ascii("".to_string())
+                    } else {
+                        return Err(TiffError::FormatError(TiffFormatError::InvalidTag));
+                    }
                 }
-                (Type::SSHORT, 3) => {
-                    let mut r = self.r(bo);
-                    Some(List(vec![
-                        Signed(i32::from(r.read_i16()?)),
-                        Signed(i32::from(r.read_i16()?)),
-                        Signed(i32::from(r.read_i16()?)),
-                    ]))
+                Type::LONG8 => {
+                    decoder.goto_offset(self.r(bo).read_u32()?)?;
+                    UnsignedBig(decoder.read_long8()?)
                 }
-                (Type::SHORT, 4) => {
-                    let mut r = self.r(bo);
-                    Some(List(vec![
-                        Unsigned(u32::from(r.read_u16()?)),
-                        Unsigned(u32::from(r.read_u16()?)),
-                        Unsigned(u32::from(r.read_u16()?)),
-                        Unsigned(u32::from(r.read_u16()?)),
-                    ]))
+                Type::DOUBLE => {
+                    decoder.goto_offset(self.r(bo).read_u32()?)?;
+                    Double(decoder.read_double()?)
+                }
+                Type::RATIONAL => {
+                    decoder.goto_offset(self.r(bo).read_u32()?)?;
+                    Rational(decoder.read_long()?, decoder.read_long()?)
+                }
+                Type::SRATIONAL => {
+                    decoder.goto_offset(self.r(bo).read_u32()?)?;
+                    SRational(decoder.read_slong()?, decoder.read_slong()?)
+                }
+                Type::__NonExhaustive => unreachable!(),
+            });
+        }
+
+        // Case 3: There is more than one value, but it fits in the offset field.
+        if value_bytes <= 4 || decoder.bigtiff && value_bytes <= 8 {
+            match self.type_ {
+                Type::BYTE => return Ok(offset_to_bytes(self.count as usize, self)?),
+                Type::SBYTE => return Ok(offset_to_sbytes(self.count as usize, self)?),
+                Type::ASCII => {
+                    let mut buf = vec![0; self.count as usize];
+                    self.r(bo).read_exact(&mut buf)?;
+                    let v = String::from_utf8(buf)?;
+                    let v = v.trim_matches(char::from(0));
```

Comment:
Interesting. This also comes from the previous implementation but reviewing it now it seems to be at odds with the validating of the one-byte case. It does _not_ validate the the last byte is indeed NUL and also removes NUL from both sides of the string. Not blocking this PR but we should open an issue on it?